File: app/src/main/java/com/wstxda/gsl/MusicSearchActivity.java
package com.wstxda.gsl;

import android.app.Activity;
import android.content.ActivityNotFoundException;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.widget.Toast;

public class MusicSearchActivity extends Activity {
    private static final String TAG = "MainActivity";

    @Override // android.app.Activity
    public void onCreate(Bundle bundle) {
        super.onCreate(bundle);
        Intent intent = new Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH");
        try {
            startActivity(intent);
        } catch (ActivityNotFoundException e) {
            Log.e(TAG, getString(R.string.unable_start) + intent, e);
            Toast.makeText(getApplicationContext(), R.string.activity_not_found, 1).show();
        }
        finish();
    }
}

File: app/src/main/java/com/wstxda/gsl/MusicSearchActivity.kt
package com.wstxda.gsl

import android.app.Activity
import android.content.ActivityNotFoundException
import android.content.Intent
import android.os.Bundle
import android.widget.Toast

class MusicSearchActivity : Activity() {

    override fun onCreate(bundle: Bundle?) {
        super.onCreate(bundle)

        try {
            val intent = Intent("com.google.android.googlequicksearchbox.MUSIC_SEARCH")
            startActivity(intent)
        } catch (e: ActivityNotFoundException) {
            handleMusicSearchNotFound()
        } finally {
            finish()
        }
    }

    private fun handleMusicSearchNotFound() {
        showToast(R.string.google_not_found)
    }

    private fun showToast(resId: Int) {
        Toast.makeText(applicationContext, resId, Toast.LENGTH_LONG).show()
    }
}